xrpl/utils/txn_parser/utils/balance_parser.py

```diff
@@ -0,0 +1,161 @@
+"""Helper functions for balance parser."""
+
+from decimal import Decimal
+from typing import Callable, List, Optional
+
+from pydash import group_by  # type: ignore
+
+from xrpl.utils.txn_parser.utils.nodes import NormalizedNode
+from xrpl.utils.txn_parser.utils.types import Balance, BalanceChange, BalanceChanges
+from xrpl.utils.xrp_conversions import drops_to_xrp
+
+
+def _get_xrp_quantity(
+    node: NormalizedNode,
+    value_parser: Callable[[NormalizedNode], Optional[Decimal]],
+) -> Optional[BalanceChange]:
+    value = value_parser(node)
+    if value is None:
+        return None
+    value_is_negative = value < Decimal(0)
+    value = drops_to_xrp(str(value.copy_abs()))
+    if value_is_negative:
+        value = Decimal(f"-{value}")
+    final_fields = node.get("FinalFields")
+    new_fields = node.get("NewFields")
+    if final_fields is not None:
+        account = final_fields.get("Account")
+        if account is not None:
+            return BalanceChange(
+                account=account,
+                balance=Balance(  # type: ignore
+                    currency="XRP",
+                    value=f"{value.normalize():f}",
+                ),
+            )
+    if new_fields is not None:
+        account = new_fields.get("Account")
+        if account is not None:
+            return BalanceChange(
+                account=account,
+                balance=Balance(  # type: ignore
+                    currency="XRP",
+                    value=f"{value.normalize():f}",
+                ),
+            )
+    return None
+
+
+def _flip_trustline_perspective(balance_change: BalanceChange) -> BalanceChange:
+    negated_balance = Decimal(balance_change["balance"]["value"]).copy_negate()
+    balance = balance_change.get("balance")
+    assert balance is not None
+    issuer = balance.get("issuer")
+    assert issuer is not None
```

Comment:
```suggestion
    balance = balance_change["balance"]
    issuer = balance["issuer"]
```
The call will automatically error if it doesn't exist